htdocs/class/file/folder.php

```diff
@@ -111,7 +111,9 @@ public function __construct($path = '', $create = true, $mode = false)
             $this->create($path, $this->mode);
         }
         if (!$this->isAbsolute($path)) {
-            $path = realpath($path);
+            $path = $this->realpath($path);
+            if (false===$path)
+                throw new Exception($path . ' not found');
```

Comment:
To provide more accurate exception handling, it is recommended to avoid \Exception, and use either custom exception classes, or one of the [SPL Exceptions](http://php.net/manual/en/spl.exceptions.php).

i.e. `throw new \InvalidArgumentException($path . ' not found');`